lib/sqlStoreStategy.js

```diff
@@ -0,0 +1,107 @@
+
+class SQLStoreStategy {
+
+  constructor(pool, table) {
+    this.pool = pool;
+    this.table = table;
+    this.dbValidated = false;
+  }
+
+  async validateDB() {
+    if (this.dbValidated) return true;
+    await this.pool.query(
+      `CREATE TABLE IF NOT EXISTS ${this.table} ` +
+      '(id varchar primary key, value varchar);'
+    );
+    this.dbValidated = true;
+    return true;
+  }
+
+  validateKey(key) {
+    if (key && key.length === 0) {
+      throw new Error('Datastore: Invalid empty key');
+    }
+  }
+
+  validateValue(value) {
+    if (value === undefined) {
+      throw new Error('Datastore: Invalid value: undefined');
+    }
+  }
+
+  async del(key) {
+    await this.validateDB();
+    this.validateKey(key);
+    await this.pool.query(`DELETE FROM ${this.table} WHERE id = $1`, [key]);
+  }
+
+  async get(key) {
+    await this.validateDB();
+    this.validateKey(key);
+    const res = await this.pool.query(
+      `SELECT value FROM ${this.table} WHERE id = $1`, [key]);
+    return res.rowCount === 0 ? undefined : JSON.parse(res.rows[0].value);
+  }
+
+  async set(key, value) {
+    await this.validateDB();
+    this.validateKey(key);
+    this.validateValue(value);
+    await this.pool.query(
+      `INSERT INTO ${this.table}(id, value) VALUES($1, $2) ` +
+      'ON CONFLICT(id) DO UPDATE SET value = $2',
+      [key, JSON.stringify(value)],
+    );
+    return value;
+  }
+
+
+  async update(key, updater) {
+    await this.validateDB();
+    this.validateKey(key);
+
+   
+    const conn = await this.pool.connect();
+
+    try {
+      await conn.query('BEGIN');
+
+      const res = await conn.query(
+        `SELECT value FROM ${this.table} WHERE id = $1 FOR UPDATE`,
```

Comment:
I found "FOR UPDATE" in a PostgreSQL article. If we want to support AnyDB we need to verify that this works in any SQL database. Nimo might know.

Reply:
I we can revisit this later

Reply:
Generally speaking this will work with Postgres, MySQL (Innodb anyway), Oracle, DB2... 
The only strange behaviour will be with (of course) SQL Server - but only in edge cases.